exec/mailinglist.R

```diff
@@ -51,56 +47,62 @@ if(arguments[["tabulate"]] & arguments[["help"]]){
+} else if (arguments[["download"]] & arguments[["modmbox"]] & arguments[["help"]]) {
+  cli::cli_alert_info("Downloads mailing list archives from mod_mbox using download_mod_mbox().")
+} else if (arguments[["download"]] & arguments[["modmbox"]]) {
 
   conf_path <- arguments[["<project_conf.yml>"]]
-  save_path <- arguments[["<save_file_name_path>"]]
-  conf <- yaml::read_yaml(conf_path)
+  start_year_month <- arguments[["<start_year_month>"]]
+  end_year_month <- arguments[["<end_year_month>"]]
+  save_folder_path <- arguments[["<save_folder_path>"]]
 
-  mod_mbox_url <- conf[["mailing_list"]][["domain"]]
-  mailing_list <- conf[["mailing_list"]][["list_key"]][1]
-
-  start_year <- arguments[["<start_year>"]]
-  end_year <- arguments[["<end_year>"]]
+  conf <- yaml::read_yaml(conf_path)
+  mailing_list <- conf[["mailing_list"]][["mod_mbox"]][["project_key_1"]][["mailing_list"]]
 
-  mbox <- download_mod_mbox(base_url = mod_mbox_url,
-                            mailing_list = mailing_list,
-                            from_year=start_year,
-                            to_year=end_year,
-                            save_file_path = save_path,
-                            verbose = TRUE)
+  download_mod_mbox(
+    mailing_list = mailing_list,
+    start_year_month = start_year_month,
+    end_year_month = end_year_month,
+    save_folder_path = save_folder_path,
+    verbose = TRUE
+  )
 
-  cli_alert_success(paste0("Downloaded mailing list was saved at: ",save_path))
-}else if(arguments[["download"]] & arguments[["modmboxmonth"]]){
+  cli::cli_alert_success(paste0("Downloaded mailing list archives were saved at: ", save_folder_path))
 
+} else if (arguments[["download"]] & arguments[["pipermail"]] & arguments[["help"]]) {
+  cli::cli_alert_info("Downloads mailing list archives from pipermail using download_pipermail().")
+} else if (arguments[["download"]] & arguments[["pipermail"]]) {
 
   conf_path <- arguments[["<project_conf.yml>"]]
-  save_path <- arguments[["<save_folder_name_path>"]]
-  conf <- yaml::read_yaml(conf_path)
-
-  mod_mbox_url <- conf[["mailing_list"]][["domain"]]
-  mailing_list <- conf[["mailing_list"]][["list_key"]][1]
+  start_year_month <- arguments[["<start_year_month>"]]
+  end_year_month <- arguments[["<end_year_month>"]]
+  save_folder_path <- arguments[["<save_folder_path>"]]
 
-  start_year <- arguments[["<start_year>"]]
-  end_year <- arguments[["<end_year>"]]
-
-  mbox <- download_mod_mbox_per_month(base_url = mod_mbox_url,
-                            mailing_list = mailing_list,
-                            from_year=start_year,
-                            to_year=end_year,
-                            save_folder_path = save_path,
-                            verbose = TRUE)
-
-  cli_alert_success(paste0("Downloaded mailing list was saved at: ",save_path))
+  conf <- yaml::read_yaml(conf_path)
+  mailing_list <- conf[["mailing_list"]][["pipermail"]][["project_key_1"]][["mailing_list"]]
+
+  download_pipermail(
```

Comment:
refresh missing here too. if the download is the refresh itself, need to separate the download of one file in one function that is being called from refresh (i believe, check with me on call. it should mimic how jira api does its thing as explained on its notebook for consistency and sanity)